RadioController.py

```diff
@@ -101,157 +109,302 @@ def parse_data(self, type_id, byte_list, length):
+    packetTypeToParser: Dict[int, Callable[[list, int], Dict[any, any]]] = {  # TODO review this type hint
+        SubpacketEnum.STATUS_PING.value: statusPing,
         SubpacketEnum.MESSAGE.value: message,
         SubpacketEnum.EVENT.value: event,
         SubpacketEnum.CONFIG.value: config,
-        # SubpacketEnum.SINGLE_SENSOR.value: single_sensor, # Todo make this a range of keys?
+        # SubpacketEnum.SINGLE_SENSOR.value: single_sensor,  # See loop that maps function for range of ids below
         SubpacketEnum.GPS.value: gps,
-        # SubpacketEnum.ACKNOWLEDGEMENT.value: acknowledgement, # TODO Uncomment when complete
+        # SubpacketEnum.ACKNOWLEDGEMENT.value: acknowledgement,
         SubpacketEnum.BULK_SENSOR.value: bulk_sensor,
     }
+    for i in SubpacketIDs.get_list_of_sensor_IDs():
+        packetTypeToParser[i] = single_sensor
 
-    # TODO review this for data type fixes
-    def fourtofloat(self, bytes):
+    # TODO Put these in utils folder/file?
+    def fourtofloat(self, byte_list):
         """
 
         :param bytes:
         :type bytes:
         :return:
         :rtype:
         """
-        assert len(bytes) == 4
-        data = bytes
+        assert len(byte_list) == 4
+        data = byte_list
         b = struct.pack('4B', *data)
         c = struct.unpack('>f' if self.bigEndianFloats else '<f', b)
         return c[0]
 
-    def fourtoint(self, bytes):
+    def fourtoint(self, byte_list):
         """
 
         :param bytes:
         :type bytes:
         :return:
         :rtype:
         """
-        assert len(bytes) == 4
-        data = bytes
+        assert len(byte_list) == 4
+        data = byte_list
         b = struct.pack('4B', *data)
         c = struct.unpack('>I' if self.bigEndianInts else '<I', b)
         return c[0]
+
+    def eighttodouble(self, byte_list):
+        """
+
+        :param bytes:
+        :type bytes:
+        :return:
+        :rtype:
+        """
+        assert len(byte_list) == 8
+        data = byte_list
+        b = struct.pack('8B', *data)
+        c = struct.unpack('>d' if self.bigEndianInts else '<d', b)
+        return c[0]
+
+# Helper class. python way of doing ++ (unlimited incrementing) TODO Put this in utils folder/file?
+class Count:
+    def __init__(self, start=0, interval=1):
+        self.interval = interval
+        self.num = start
+
+    def __iter__(self):
+        return self
+
+    def curr(self):
+        return self.num
+
+    # increments and returns the new value
+    def next(self, interval=0):
+        if interval == 0:
+            self.num += self.interval
+        else:
+            self.num += interval
+
+        return self.num
+
+    # returns the current value then increments
+    def currAndInc(self, interval=9):
+        num = self.num
+        if interval == 0:
+            self.num += self.interval
+        else:
+            self.num += interval
+
+        return num
```

Comment:
Definitely belongs in detail.py or something similar, IMO.

Reply:
Moved. Less obvious is bitFromByte. I'm leaving it in for now